swim/memberlist.go

```diff
@@ -324,3 +324,17 @@ func (m *memberlist) String() string {
 func (m *memberlist) Iter() *memberlistIter {
 	return newMemberlistIter(m)
 }
+
+func (m *memberlist) GetActiveMemberAddresses() []string {
+	var active []string
+
+	m.members.RLock()
```

Comment:
Is it necessary to have a read-lock while we append to the active list? I'm thinking that we can make the locking more fine-grained by simply locking while copying `m.members.list` and iterate over the copy while not locking.